src/parser.ts

```diff
@@ -158,12 +158,12 @@ export class SystemVerilogParser {
 
     private r_params: RegExp = new RegExp(
         [
-            /(?<type>parameter)\s+/,
-            /(?:(?<data_type>\w+\s*)*?)/,
+            /(?<type>parameter)/,
+            /(?:(\s+(?<data_type>\w+))*)/,
             // Multi-dimensional arrays
-            /(?:\[.*?\]\s*)*?/,
+            /(?:\s*\[.*?\]\s*)*?/,
             // Multiple declaration not allowed (yet)
-            /(?<name>\w+)\s*/
+            /\s+(?<name>\w+)\s*/
```

Comment:
Is whitespace guaranteed here? I think it should be optional, right?
```suggestion
            /\s*(?<name>\w+)\s*/
```

Reply:
I think it is necessary to require whitespace before the name. If not, it would match when there is no whitespace between the parameter and name, e.g `parametername`. It would match `parameter` and `name` into the correct group, but I would guess that this would be invalid SystemVerilog syntax. 

I am not able to come up with a parameter declaration which does not have whitespace before the name. Do you know of any examples of this? Not sure if both `parameter type [1:0]name` and `parameter type [1:0] name` are valid, or just the latter. In the end, I think both solutions will work for the purpose of the extension, but not requiring whitespace will match some additional and potentially invalid declarations.

Reply:
OK, I'm fine with this implementation.